src/expr.cpp

```diff
@@ -1446,14 +1458,43 @@ SubscriptExpr::SubscriptExpr(std::shared_ptr<Expr> _arr,
     evaluate(ctx);
 }
 
-void SubscriptExpr::setValue(std::shared_ptr<Expr> _expr) {
+void SubscriptExpr::setValue(std::shared_ptr<Expr> _expr,
+                             std::deque<size_t> &span,
+                             std::deque<size_t> &steps) {
+    EvalCtx ctx;
+    auto eval_res = idx->evaluate(ctx);
+    // TODO: step can be negative
+    size_t step = 0;
+    bool is_idx_iter = eval_res->isIterator();
+    if (is_idx_iter) {
+        auto iter_eval_res = std::static_pointer_cast<Iterator>(eval_res);
+        eval_res = iter_eval_res->getEnd()->evaluate(ctx);
+        auto step_eval = iter_eval_res->getStep()->evaluate(ctx);
+        if (!step_eval->isScalarVar())
+            ERROR("only scalar variables are supported for now");
+        auto step_scalar_val = std::static_pointer_cast<ScalarVar>(step_eval);
+        step = step_scalar_val->getCurrentValue()
+                   .castToType(IntTypeID::ULLONG)
+                   .getValueRef<size_t>();
```

Comment:
`size_t` maps to `unsigned long` on macOS, which is not implemented. So it causes build error:
```c++
Undefined symbols for architecture arm64:
  "unsigned long& yarpgen::IRValue::getValueRef<unsigned long>()", referenced from:
      yarpgen::SubscriptExpr::setValue(std::__1::shared_ptr<yarpgen::Expr>, std::__1::deque<unsigned long, std::__1::allocator<unsigned long> >&, std::__1::deque<unsigned long, std::__1::allocator<unsigned long> >&) in libyarpgen_lib.a(expr.cpp.o)
```